internal/aggregated-co/writer.go

```diff
@@ -0,0 +1,66 @@
+package aggregated_co
+
+import (
+	"context"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// From https://github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorstatus/writer.go
+
+// NewWriter returns a new instance of Writer.
+func NewWriter(client configv1client.ConfigV1Interface) *Writer {
+	return &Writer{
+		client: client,
+	}
+}
+
+// Writer encapsulates logic for cluster operator object API. It is used to
+// update ClusterOperator resource.
+type Writer struct {
+	client configv1client.ConfigV1Interface
+}
+
+// EnsureExists ensures that the cluster operator resource exists with a default
+// status that reflects expecting status.
+func (w *Writer) EnsureExists(name string) (existing *configv1.ClusterOperator, err error) {
```

Comment:
Always pass a context in from the function signature if you ever need to call anything that accepts a context.
```suggestion
func (w *Writer) EnsureExists(ctx context.Context, name string) (existing *configv1.ClusterOperator, err error) {
```